### PR TITLE
Add support for static resource requests

### PR DESCRIPTION
**Pull request by ScheererJ**
**How to categorize this PR?**

/area networking
/area auto-scaling
/kind enhancement

**What this PR does / why we need it**:

Add support for static resource requests.

While it is generally best practice to utilise automatic resource management and scaling in kubernetes environments there may be situations when the workload is static or any disruptions may cause serious issues. In such cases it is better to be able to use static resource allocation instead of dynamic autoscaling like cluster-proportional or vertical pod autoscaler. This is in particular true for system components such as calico, which may cause issues in the pod network if a process is not available for an extended amount of time.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
Add static resource allocation autoscaling mode for calico node/typha (autoScaling.mode: static).
```

### CODE REVIEW

File: pkg/apis/calico/types_network.go [hunk 1]
```diff
@@ -155,10 +156,22 @@ const (
 	AutoscalingModeClusterProportional AutoscalingMode = "cluster-proportional"
 	// AutoscalingModeVPA is a constant for vertical pod autoscaling mode.
 	AutoscalingModeVPA AutoscalingMode = "vpa"
+	// AutoscalingModeStatic is a constant for static resource allocation as autoscaling mode.
+	AutoscalingModeStatic AutoscalingMode = "static"
 )
 
 // AutoScaling defines how the calico components are automatically scaled. It allows to use either vertical pod or cluster-proportional autoscaler (default: cluster-proportional).
 type AutoScaling struct {
 	// Mode defines how the calico components are automatically scaled. It allows to use either vertical pod or cluster-proportional autoscaler (default: cluster-proportional).
```
Comment (axel7born) [inline pkg/apis/calico/types_network.go:165]: `static` mode should be added to this comment.

File: docs/usage/usage.md [hunk 2]
```diff
@@ -81,6 +80,25 @@ autoScaling:
...
+  resources:
+    node:
+      cpu: 100m
+      memory: 100Mi
+    typha:
+      cpu: 100m
+      memory: 100Mi
+```
+
+> ℹ️ Please note that it is only possible to (optionally) configure the resource requests for `calico-node` and `calico-typha` in static mode.
```
Comment (axel7born) [inline docs/usage/usage.md:99]: I'm not sure if this sentence is really understandable. At least, I got it wrong reading it for the first time.
It is not needed, to configure both `calico-node` **and** `calico-typha` and the `optionally` means that if resources are not specified there is a default.
Maybe, we should use the default values in this example.
suggestion:
```
"Please note that in static mode, you have the option to configure the resource requests for calico-node and calico-typha. If not specified, default settings will be used."
```

**Comment by axel7born** [comment]
The PR looks good to me. I have only small suggestions regarding docs or comments in the code.

**Review by axel7born — CHANGES_REQUESTED** [review]

File: docs/usage/usage.md [hunk 3]
```diff
@@ -65,14 +65,13 @@ spec:
 
 ## AutoScaling
 
-Autoscaling defines how the calico components are automatically scaled. It allows to use either vertical pod or cluster-proportional autoscaler (default: cluster-proportional).
+Autoscaling defines how the calico components are automatically scaled. It allows to use either static resource assignment, vertical pod or cluster-proportional autoscaler (default: cluster-proportional).
 
-The cluster-proportional autoscaling mode is preferable when conditions require minimimal disturbances and vpa mode for improved cluster resource utilization. 
+The cluster-proportional autoscaling mode is preferable when conditions require minimal disturbances and vpa mode for improved cluster resource utilization. Static resource assignments causes the least disruptions, but has no dynamics to handle changing demands.
```
[the remark this answers is not in the document]
Comment (ScheererJ) [reply docs/usage/usage.md]: Thanks for the simplification. Added with https://github.com/gardener/gardener-extension-networking-calico/pull/464/commits/d66173fd499a37fa5a80f927d5fe0cfa0a719f2d.

File: pkg/apis/calico/types_network.go (hunk 1 above)
Comment (ScheererJ) [reply pkg/apis/calico/types_network.go]: Adapted the documentation strings of corresponding fields/types in https://github.com/gardener/gardener-extension-networking-calico/pull/464/commits/d66173fd499a37fa5a80f927d5fe0cfa0a719f2d.

File: docs/usage/usage.md (hunk 2 above)
Comment (ScheererJ) [reply docs/usage/usage.md]: Adapted the section to hopefully be more clear.
WDYT?

**Review by axel7born — APPROVED** [review]

**Pull request merged**